storage-operator/cmd/manager/main.go

```diff
@@ -80,19 +79,14 @@ func main() {
 		os.Exit(1)
 	}
 
-	ctx := context.TODO()
-	// Become the leader before proceeding
-	err = leader.Become(ctx, "storage-operator-lock")
-	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
 	// Create a new Cmd to provide shared dependencies and start components
 	mgr, err := manager.New(cfg, manager.Options{
-		Namespace:          namespace,
-		MapperProvider:     restmapper.NewDynamicRESTMapper,
-		MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+		Namespace:               namespace,
+		MapperProvider:          restmapper.NewDynamicRESTMapper,
+		MetricsBindAddress:      fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+		LeaderElection:          true,
+		LeaderElectionNamespace: "kube-system",
+		LeaderElectionID:        "storage-operator-lock",
```

Comment:
We may want to configure `LeaseDuration` (default 15s) and `RenewDeadline` (default 10s) values if we have an idea of what a "right" value would be.
Otherwise let's keep the default for now…

See https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/manager#Options

Reply:
I guess the default values are fine as they are.

Reply:
I think so, we will see when it more widely use if we need to tailor it

Reply:
We run a single-replica *Deployment* anyway... (which can in some circumstances lead to multiple replicas running concurrently, but anyway).

10s renewal, 15s lease (i.e., 'only' 5s diff) *can* be an issue when a system is under high churn, the process has unreasonably long GC pauses (could be caused by other processes on the same image), when running in VMs on a heavily over-allocated host,...

I guess indeed, we'll see how it plays out in practice. We *must* make sure the whole system can work without data loss/corruption in case multiple operators would run concurrently anyway, i.e. fully idempotent Salt states/actions (slightly concerned about the 'forced' formatting of `sparseLoopDevice`s here), handling conflicting updates to the API-server gracefully (I think we got that covered),...

Reply:
Yup, the conflicting updates are covered, we have some log message about it and we have automatic retry for those.